cinn/hlir/framework/graph_compiler.h

```diff
@@ -132,6 +133,18 @@ class GraphCompiler final {
   // and erase them from the scope to avoid unnecessary buffer allocation
   void RemoveInvalidVariables(const std::vector<std::unique_ptr<Instruction>>& instructions);
 
+  // find the first and last instruction where a variable used, and mark the
+  // variable should allocate buffer before the first instruction runing and
+  // can release the buffer after the last instruction finished.
+  void AnalysisVariableUsedLife(const std::vector<std::unique_ptr<Instruction>>& instructions,
```

Comment:
private method ?

Reply:
Yes, it has been declared under `private`.